buildpipeline/pipeline.json

```diff
@@ -16,109 +16,151 @@
       },
       "Definitions": [
         {
-          "Name": "Core-Setup-CentOS-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "centos.7"
+          },
           "ReportingParameters": {
             "OperatingSystem": "CentOS 7.1",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "Core-Setup-Debian8-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "debian.8",
+            "REPO_ID": "579f8fb0fedca9aeeb399132",
+            "REPO_USER": "dotnet",
+            "REPO_PASS": "PassedViaPipeBuild",
+            "REPO_SERVER": "azure-apt-cat.cloudapp.net"
+          },
           "ReportingParameters": {
             "OperatingSystem": "Debian 8.2",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "Core-Setup-Fedora23-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "fedora.23"
+          },
           "ReportingParameters": {
             "OperatingSystem": "Fedora 23",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "Core-Setup-Fedora24-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "fedora.24"
+          },
           "ReportingParameters": {
             "OperatingSystem": "Fedora 24",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "Core-Setup-OpenSuse42.1-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "opensuse.42.1"
+          },
           "ReportingParameters": {
             "OperatingSystem": "OpenSuse 42.1",
             "Type": "build/product/",
             "Platform": "x64"
           }
         },
         {
-          "Name": "Core-Setup-OSX-x64",
+          "Name": "Core-Setup-Linux",
+          "Parameters": {
+            "PB_DockerOS": "ubuntu.14.04",
+            "REPO_ID": "562fbfe0b2d7d0e0a43780c4",
+            "REPO_USER": "dotnet",
+            "REPO_PASS": "PassedViaPipeBuild",
```

Comment:
Sounds good. If we could filter which pipeline variables go to which builds, then something like this definitely would be valuable, but not yet. (That would have even more value in a case like CoreFX where the NuGet API key is only needed by a single build leg.)